app/jobs/user_assignments/project_group_un_assignment_job.rb

```diff
@@ -32,5 +33,27 @@ def remove_users_from_my_modules(project, users)
       my_modules.update_all(updated_at: Time.current)
       # rubocop:enable Rails/SkipsModelValidations
     end
+
+    def remove_users_from_projects(project, users)
+      users.each do |user|
+        user_assignment = project.user_assignments.find_by(user: user)
+
+        if user_assignment.last_with_permission?(ProjectPermissions::USERS_MANAGE, assigned: :manually)
+          raise ActiveRecord::RecordInvalid
+        end
+
+        propagate_job(project, user_assignment, destroy: true)
+      end
+    end
+
+    def propagate_job(project, user_assignment, destroy: false)
```

Comment:
why do we need to propagate if we already removed the user from tasks and experiments?

Reply:
i tried removing assignments the same way it is for experiments/modules but then it just stuck in a loop, im not sure why, but this way works 100%